sdk/core/core-http/src/credentials/accessTokenRefresher.ts

```diff
@@ -0,0 +1,67 @@
+import { AccessToken, TokenCredential, GetTokenOptions } from "@azure/core-auth";
+
+/**
+ * Helps the core-http token authentication policies with requesting a new token if we're not currently waiting for a new token.
+ */
+export class AccessTokenRefresher {
```

Comment:
After looking over the code I wonder if we even need a separate class for this behavior.  You could just do the promise management inside of `BearerTokenAuthenticationPolicy` itself.

Reply:
I think that @schaabs could help me here. The idea is to be able to have something plug-able, to use somewhere else, as far as I understand.

Reply:
I spoke with @schaabs on Teams, he clarified things for me.